### PR TITLE
github/workflows: add merge when ready workflow

### PR DESCRIPTION
**Pull request by arnout**
Using [PR merge bot](https://github.com/marketplace/actions/pr-merge-bot), automatically merge a pull request when the
checks pass and the MergeWhenReady tag is set.

### CODE REVIEW

**Comment by arnout** [comment]
The output from merge bot is not very promising:
```
Run squalrus/merge-bot@v0.2.0
  with:
    GITHUB_TOKEN: ***
    labels: MergeWhenReady
    blocking-labels: don't merge
    checks: true
    method: rebase
    test: false
    reviewers: true
test mode: false
required reviewers: true
required labels: ["MergeWhenReady"]
blocking labels: ["don't merge"]
merge method: rebase
##[error]Not Found
##[error]Node run failed with exit code 1
```

I expect the action to fail, but not with "Not found"...

File: .github/workflows/merge-when-ready.yml
Comment (rmelotte) [inline .github/workflows/merge-when-ready.yml]: Why "v0.2.0"? Is it intentional? It looks like the latest one is "v0.4.0"

Comment (arnout) [reply .github/workflows/merge-when-ready.yml]: He, I just copy-pasted from the example... Will fix.

**Review by rmelotte — COMMENTED** [review]
[Just adding a review comment to test the bot.]

**Comment by rmelotte** [comment]
It looks like checks do not include the reviews.

**Comment by arnout** [comment]
> It looks like checks do not include the reviews.

Darn. I assumed that the `checks: true` part would cover the reviews.

I intentionally turned the reviews off because according to the documentation says "This enforces a reviewer mode where there cannot be any pending reviews". This basically means that *all* code owners must review.

I'll try again with reviews and when we get two accepts we can check if it works.

**Comment by rmelotte** [comment]
Yes the behavior with the reviews doesn't exactly match what we used to do.
It also seems that if it has run once on one commit, it won't run again when the other checks completed (which defeats the purpose here, since we want it to run when the CI has finished).

It's not perfect, but if we could at least make it run one time /after/ all the other checks completed, we could already improve our current workflow:
- when you are certain that the PR is ready to be merged, rebase and add a tag.
- with the tag, the merge bot runs once CI has finished and merge it if it passed.

If I understand the way it works correctly, it wouldn't work if you retry a failed pipeline for example.

**Review by rmelotte — APPROVED** [review]

**Comment by arnout** [comment]
Since it doesn't get triggered when CI finishes, I don't think this is useful. So I'm closing this PR.

**Pull request closed without merging**
